Drupal 7.x civicrm 4.4 Security Warning

khyjb1995 opened the pull request:
Problem1. ||The CiviCRM debug log should not be downloadable.Read more about this warning||
Problem2. ||The upload & custom directory file should not be downloadable ||   if set permission to 755 also like that if move out webroot then it display ""Notice：Undefined offset: 1 at CRM_Utils_Check_Security->guessUrl() (/var/www/html/sites/all/modules/civicrm/CRM/Utils/Check/Security.php in line 290)。""

eileenmcnaughton committed:
CRM-14251 views integration:event links - return empty instead of non-link
text where link is not permissioned

Commit:
As discussed in CRM-14267, applied patch and tested.

eileenmcnaughton committed:
CRM-14437 CiviEngage walklist fixes

atif-shaikh committed:
CRM-14821

xurizaemon committed:
Fix for CRM-14389. Support Float data in CiviCRM Views

xurizaemon committed:
CRM-14936. Check for existence of Block module.

Commit:
CRM-15074 - _civicrm_registerClassLoader() should store a relative path.

Commit:
Use UFGroup name instead of title in Drupal profile url

colemanw committed:
CRM-15337 - Fix undefined index in civicrm_metatag_page_cache_cid_parts_alter

Conflicts:
	civicrm.module

Commit:
CRM-15425: patched views field handlers for file URL output

eileenmcnaughton committed:
Merge pull request #186 from eileenmcnaughton/7.x-4.4

CRM-15425: patched views field handlers for file URL output

eileenmcnaughton committed:
Merge pull request #195 from eileenmcnaughton/CRM-15337

CRM-15337 - Fix undefined index in civicrm_metatag_page_cache_cid_parts_...

eileenmcnaughton committed:
Merge pull request #196 from eileenmcnaughton/CRM-14251

various minor fixes

eileenmcnaughton committed:
Merge pull request #197 from eileenmcnaughton/4.4.11

4.4.11

eileenmcnaughton committed:
Merge pull request #198 from eileenmcnaughton/7.x-4.4

CRM-14936. Check for existence of Block module.

Commit:
Issue CRM-15660. Use CiviCRM custom field handler to render NULL as an empty field.

eileenmcnaughton committed:
Merge pull request #200 from xurizaemon/7.x-4.4-CRM-15660

Issue CRM-15660. Use CiviCRM custom handler to render NULL as empty string

davejenx committed:
CRM-15644 - fix civicrm_group table missing error on login when CiviCRM DB is separate from Drupal's

----------------------------------------
* CRM-15644: civicrm_group table "missing" error on login when CiviCRM DB is separate from Drupal's
  https://issues.civicrm.org/jira/browse/CRM-15644

eileenmcnaughton committed:
CRM-15140 - port further metadata fixes

eileenmcnaughton committed:
Merge pull request #209 from eileenmcnaughton/7.x-4.4

CRM-15140 - port further metadata fixes

eileenmcnaughton commented:
This commit is not 'clean'

eileenmcnaughton committed:
Merge pull request #201 from davejenx/CRM-15644

CRM-15644 - fix civicrm_group table missing error on login when CiviCRM ...

Commit:
Fix the title of user profile page

Commit:
Adding translation for profile menu tab

Commit:
Adding translation for profile menu tab

eileenmcnaughton commented:
In order to clean this up for merging someone should either

git pull --rebase origin/7.x-4.4

or start from a fresh 7.x-4.4 & cherry-pick the commit in.

I'm not sure what JIRA ticket this PR relates to

eileenmcnaughton committed:
CRM-15871 fix urls for metatag & language switcher

eileenmcnaughton committed:
Merge pull request #214 from eileenmcnaughton/7.x-4.4

CRM-15871 fix urls for metatag & language switcher

eileenmcnaughton committed:
CRM-15330 in order to support existing urls use title as name if in the url

eileenmcnaughton committed:
CRM-15330 add more protection for existing urls

eileenmcnaughton committed:
Merge pull request #222 from eileenmcnaughton/7.x-4.4

CRM-15330 - add support for existing title based urls

eileenmcnaughton committed:
CRM-15330 fix merge error

eileenmcnaughton committed:
Merge pull request #224 from eileenmcnaughton/7.x-4.4-11

CRM-15330 second PR

eileenmcnaughton committed:
CRM-15952 fix for edit screen mishandling profiles

eileenmcnaughton committed:
Merge pull request #226 from eileenmcnaughton/CRM-15952

Adding translation for profile menu tab

eileenmcnaughton committed:
Merge pull request #227 from eileenmcnaughton/7.x-4.4-12

CRM-15952 fix for edit screen mishandling profiles

eileenmcnaughton committed:
CRM-15929 improve log messages

eileenmcnaughton committed:
Merge pull request #232 from eileenmcnaughton/message

CRM-15929 improve log messages

eileenmcnaughton committed:
CRM-15952 handling for possibility of non-unique name

eileenmcnaughton committed:
Merge pull request #235 from eileenmcnaughton/7.x-4.4-CRM-15952

CRM-15952 handling for possibility of non-unique name

eileenmcnaughton committed:
stylistic synchronisation with upstream

eileenmcnaughton committed:
Merge pull request #240 from eileenmcnaughton/7.x-4.4

stylistic synchronisation with upstream

eileenmcnaughton committed:
enotice fix on drush uli screen

eileenmcnaughton committed:
Merge pull request #242 from eileenmcnaughton/7.x-4.4

enotice fix on drush uli screen

eileenmcnaughton committed:
CRm-15871 enotice fix per bgm comment

eileenmcnaughton committed:
Merge pull request #249 from eileenmcnaughton/7.x-4.4-notice

CRm-15871 enotice fix per bgm comment

xurizaemon committed:
CRM-16048. Expose custom field values as well as labels through Views.

eileenmcnaughton committed:
CRM-16050 issue with previous attempt to fix bugs from CRM-15330

eileenmcnaughton committed:
Merge pull request #253 from eileenmcnaughton/CRM-16050

CRM-16050 issue with previous attempt to fix bugs from CRM-15330

colemanw committed:
Merge pull request #252 from xurizaemon/7.x-4.4-CRM-16048

CRM-16048. Expose custom field values as well as labels through Views.

Commit:
CRM-15030, CRM-16048. Reapply CRM-15030 which was reversed by CRM-16048.

Commit:
CRM-16122. Expose custom field values in human-friendly format.

Having to do implode(explode)) feels a bit hackish ... should CiviCRM be providing some formatting on values retrieved by this point?

eileenmcnaughton committed:
Merge pull request #263 from xurizaemon/7.x-4.4-CRM-16122

CRM-16122. Expose custom field values in human-friendly format.

eileenmcnaughton committed:
Merge pull request #262 from xurizaemon/7.x-4.4-CRM-16121

CRM-15030, CRM-16048. Reapply CRM-15030 which was reversed by CRM-16048.

eileenmcnaughton committed:
CRM-16309 (drupal user edit profile) fix redirect on delete contact image byt setting context

kurund committed:
Merge pull request #276 from eileenmcnaughton/7.x-4.4

CRM-16309 (drupal user edit profile) fix redirect on delete contact imag...

Commit:
CRM-16022. Correct "project" field in CiviCRM info files.

eileenmcnaughton committed:
Merge pull request #283 from xurizaemon/7.x-4.4-CRM-16022

7.x-4.4 CRM-16022 fix project info

eileenmcnaughton commented:
I'm going to close this on the basis the PR is so 'dirty' it is not helpful keeping it open. Last updated in Jan

I think it would be worth following up on the issue referenced - but we need a cherry-pick of the relevant commit - probably against 7.x-master
